### DISCUSSION TITLE
form validation failed if mandatory field is hidden in the table

### DISCUSSION THREAD

**Discussion by 549393092** [quasarframework/quasar#12089] [General - Components / Directives / etc]
Dear Quasar team,

I did a demo to illustrate this issue, could you please have a look?
https://codepen.io/Stanley-549393092/pen/wvrZGZy?editors=1111
It validates input based on form level(this.$refs.vform.validate), if you click "validate" button, the validation is not passed because mandatory fields are not input.
However, if user hide that column by "layout" button and click "validate" button again, the validation is passed.
Is there any way to avoid that? Thanks a lot!

**Comment by pdanpdan** [comment]
If the component is not rendered then it cannot be validated.
It's not a good idea to validate based on rendered components anyway - you should validate the data/model

**Reply by 549393092** [reply]
@pdanpdan Ok, got it. Because there is a workaround for panel (q-input is hidden in another tab), please see my previous post
https://forum.quasar-framework.org/topic/7752/solved-qform-validation-doesn-t-catch-error-from-another-inactive-panel,
So I am thinking if there is some way to detect the validation result even if q-input is hidden in q-table.
If that not work, I think it's has to set mandatory fields as "fixed" column or add the check in the back-end as well.

**Comment by 549393092** [comment]
I just used another way to show/hide column with css style `<td style="display:none;">`, and it can detect error from hidden column.
So it that possible to change the way of show/hide column? Currently, I guess it's using "v-if" to show hide column, can we use "v-show" or change column style for it?
